src/components/tasks/card/index.tsx

```diff
@@ -308,11 +315,20 @@ const Card: FC<Props> = ({
         const data = {
             percentCompleted: percentCompleted,
         };
-        await updateTasks({
-            task: data,
-            id: id,
-        });
-        toast(SUCCESS, 'Progress Updated Successfully');
+        if (isUserAuthorized) {
+            await updateTask({
+                task: data,
+                id: id,
+            })
+                .unwrap()
+                .then(() => toast(SUCCESS, 'Progress Updated Successfully'))
+                .catch(() => toast(ERROR, 'Something Went wrong!'));
+        } else {
+            await updateSelfTask({ task: data, id: id })
+                .unwrap()
+                .then(() => toast(SUCCESS, 'Progress Updated Successfully'))
+                .catch(() => toast(ERROR, 'Something Went wrong!'));
```

Comment:
Can you move this error message to separate constants files?

Reply:
fixed